### PR TITLE
Close TestingIgniteServer lease 

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description

Close TestingIgniteServer lease in `IgniteQueryRunner`

## Additional context and related issues

Similar to we do in `PhoenixQueryRunner` 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java [hunk 1]
```diff
@@ -80,14 +81,16 @@ public static void main(String[] args)
     {
         Logging.initialize();
 
-        DistributedQueryRunner queryRunner = createIgniteQueryRunner(
-                TestingIgniteServer.getInstance().get(),
-                ImmutableMap.of("http-server.http.port", "8080"),
-                ImmutableMap.of(),
-                TpchTable.getTables());
+        try (SharedResource.Lease<TestingIgniteServer> instance = TestingIgniteServer.getInstance()) {
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java:84]: We can't connect the Ignite server if we close this like this.

**Review by ebyhr — CHANGES_REQUESTED** [review]

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java (hunk 1 above)
Comment (chenjian2664) [reply plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java:84]: Correct me if I am wrong, my understanding is, here is only the test for createIgniteQueryRunner method with the TestingIgniteServer, so before the main process exit we need close the resource.

Comment (ebyhr) [reply plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java:84]: This isn't a test. We use this method when we want to access Trino or Ignite server in local environment.

Comment (chenjian2664) [reply plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java:84]: ha, yes, I just notice it's only for allowing access the server in local environment.

Comment (chenjian2664) [reply plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java:84]: We should remove the close logic in `PhoenixQueryRunner`

**Pull request closed without merging**